frame/elections-phragmen/src/lib.rs

```diff
@@ -933,32 +943,24 @@ impl<T: Config> Pallet<T> {
 			Ok(_) => (),
 			Err(_) => {
 				log::error!(
-					target: "runtime::elections-phragmen",
+					target: "runtime::elections",
 					"Failed to run election. Number of voters exceeded",
 				);
 				Self::deposit_event(Event::ElectionError);
-				return T::DbWeight::get().reads(3 + max_voters as u64)
+				return T::DbWeight::get().reads(3 + max_voters as u64);
 			},
 		}
 
-		// used for phragmen.
-		let voters_and_votes = voters_and_stakes
-			.iter()
-			.cloned()
-			.map(|(voter, stake, votes)| {
-				num_edges = num_edges.saturating_add(votes.len() as u32);
-				(voter, to_votes(stake), votes)
-			})
-			.collect::<Vec<_>>();
-
 		let weight_candidates = candidates_and_deposit.len() as u32;
-		let weight_voters = voters_and_votes.len() as u32;
-		let weight_edges = num_edges;
-		let _ =
-			sp_npos_elections::seq_phragmen(num_to_elect, candidate_ids, voters_and_votes, None)
-				.map(|ElectionResult::<T::AccountId, Perbill> { winners, assignments: _ }| {
-					// this is already sorted by id.
-					let old_members_ids_sorted = <Members<T>>::take()
+		let weight_voters = voters_and_stakes.len() as u32;
+		let weight_edges = 0; // TODO(gpestana): generalize for elections
+
+		let _ = T::ElectionProvider::elect().map(|mut winners| {
+			// sort winners by stake
+			 winners.sort_by(|i, j| j.1.total.cmp(&i.1.total));
```

Comment:
Instead of explicitly sorting the winners by stake for every election, should we instead have a `SortedListProvider`? If not, should the sorting be done in the `ElectionDataProvider`?

Reply:
Unfortunately nothing in the new interface requires this to be sorted. the old interface was sorted though: https://paritytech.github.io/substrate/master/sp_npos_elections/phragmen/fn.seq_phragmen.html

I think using `Solver` might help with this as well, other than simplifying a few other things.